src/include/miopen/generic_search.hpp

```diff
@@ -296,6 +298,43 @@ using RunAndMeasure_t =
                                                           std::declval<ConvSolution>(),
                                                           std::declval<float&>()));
 
+template <class Solver, class Context>
+auto GetAllConfigs(const Solver s, const Context& context)
+    -> const ComputedContainer<decltype(s.GetPerformanceConfig(context)), Context>
+{
+    using PerformanceConfig = decltype(s.GetPerformanceConfig(context));
+
+    const ComputedContainer<PerformanceConfig, Context> main(context);
+    const int main_size = std::distance(main.begin(), main.end());
+    const ComputedContainer<PerformanceConfig, Context> spare(context, true);
+    const int spare_size = std::distance(spare.begin(), spare.end());
+    const bool useSpare  = (main_size == 0);
+
+    const ComputedContainer<PerformanceConfig, Context> all_configs = useSpare ? spare : main;
+    const int n_runs_total = useSpare ? spare_size : main_size;
+    MIOPEN_LOG_W(s.SolverDbId() << ": Searching the best solution among " << n_runs_total
+                                << (useSpare ? " (spare)" : "") << "...");
+
+    return all_configs;
+}
+
+template <class Solver, class Context>
+std::vector<ConvSolution> GetAllSolutions(const Solver s, const Context& context_)
+{
+    auto context                  = context_;
+    context.is_for_generic_search = true;
+
+    auto all_configs = GetAllConfigs(s, context);
+
+    std::vector<ConvSolution> solutions;
+    for(const auto& current_config : all_configs)
+    {
+        ConvSolution current_solution = s.GetSolution(context, current_config, true);
+        solutions.push_back(current_solution);
+    }
+    return solutions;
```

Comment:
This breaks the idea behind ComputedContainer. The point is to not occupy memory, thus making the potential number of tuning configs unlimited. That is why we are using ComputedContainer instead of a regular STL container. As soon as we copy the "virtual" content of ComputedContainer into a regular STL container, it starts to take up memory.

Reply:
This is an addition for Fin performance db compile step. What I'm trying to accomplish here is to pass the list of all solutions up to Fin so that Fin can do a precompile kernel for each solution. This also allows Fin to select each kernel created and add them to a json output file.
Perhaps there is a way to pass the solutions virtually?

Reply:
What about the following design.

Generic search checks for some callback (pointer to a function). If it is nullptr (default state), then generic search works as usual. If it is set, then, instead of building the Solution and benchmarking it, generic search calls that function (passing each Solution as a parameter).

`fin` should set the callback prior launching search. The callback may simply collect received Solutions into some container (for further handling) or immediately perform necessary actions with each Solution (compile and add to a json output).

Reply:
@cderb IMPORTANT QUESTION: What is the purpose of the compiled kernels? I am asking because the library includes only tunable kernels into Solutions when tuning is in progress. For example, if a Solution consists of two kernels (convolution + some transform) and only convolution kernel is tunable, then the library includes only convolutions into solutions for tuning.

Reply:
Interesting idea, It will take me some more development time to figure that out. I'd like to work on that in a separate PR since this method is stable currently.

The purpose of collecting the compiled kernels is entirely for Tuna's tuning process. In our compile step we store all possible kernel compilations in Tuna's mysql database, the evaluation step then consumes these kernels from mysql. It acts as a way of separating our compile machines and evaluation machines.